<?xml version="1.0"?>

<launch>
  <arg name="uri" default="radio://0/80/2M/E7E7E7E701" />
  <arg name="joy_dev" default="/dev/input/js0" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="crazyflie">
    <!--<param name="crazyflie/pid_rate/yaw_kp" value="200"/>-->
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri)" />
      <param name="tf_prefix" value="crazyflie" />
      <param name="enable_logging" value="True" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["motor.m1", "motor.m2", "motor.m3", "motor.m3"]
      </rosparam>
    </node>


    <node name="joy" pkg="joy" type="joy_node" output="screen" >
      <param name="dev" value="$(arg joy_dev)" />
    </node>

    <include file="$(find crazyflie_demo)/launch/xbox360.launch">
    </include>

    <node name="crazyflie_demo_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
    </node>
  </group>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie.rviz" />

  <!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot" args="/crazyflie/temperature/temperature"/> -->


</launch>
